### PR TITLE
Add test for the polygon locator on an ORCA grid

### PR DESCRIPTION
**Pull request by twsearle**
## Description

The atlas [polygon locator](https://github.com/ecmwf/atlas/blob/develop/src/atlas/util/PolygonLocator.h) identifies the partition that owns a target point. This doesn't work properly on ORCA grids, and this test is here to expose this weakness.

Partly this is due to the use of longitude/latitude coordinates in the polygon locator, however for ORCA grids these cannot be assumed to lie between 0-360.

This test exposes this weakness in the ORCA025 grid. Every grid apart from ORCA2 seems to fail in the same way. The ORCA2 grid will also fail, however this is a more serious issue that I am not sure how to resolve and I am not sure it is worth it at this time. I believe this is down to the identification of polygons on an ORCA2 grid. The ORCA2 grid contains a cut-out area for the mediterranean. This region has a tendency to make the partition polygon non-convex, which I believe breaks the assumptions of the polygon locator. The ORCA2 grid is very small and unlikely to need to run under MPI, however it is worth pointing this behaviour out in case it affects other grids.

An example of the orca2 grid partition polygon under equal regions:
orca2_polygon_eq_regions_fail

An example under the checkerboard partitioner:

orca2_polygon_checkerboard_fail

The underlying 2D orca2 grid:

orca2_grid

### CODE REVIEW

File: src/tests/test_orca_polygon_locator.cc [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            REQUIRE( mesh.grid() );
+            EXPECT( mesh.grid().name() == gridname );
+            atlas::util::ListPolygonXY polygon_list(mesh.polygons());
+            //mesh.polygon(0).outputPythonScript("polygon.py",
+            //                                   Config("nodes", false)("coordinates", "xy"));
+            atlas::util::PolygonLocator locator(polygon_list, mesh.projection());
+
+            // fails on ORCA2 because partition polygon is not connected space
+            // (ORCA2 grids have a cut out area for the mediterranean)
+            idx_t part = locator({82.0, 10.0});
```
Comment (wdeconinck) [inline src/tests/test_orca_polygon_locator.cc:56]: How could this be solved... ?
It would be good to have a use case to know what would be expected.

**Comment by wdeconinck** [comment]
This test has been added to develop. Thank you for this.
It would take a LOT of effort and some redesign to support partitions that are not covered by one single polygon.
The ORCA2 mesh is hopefully the only edge case, and is not so important nowadays to fix this issue.
